src/net.cpp

```diff
@@ -386,6 +386,27 @@ static CAddress GetBindAddress(const Sock& sock)
     return addr_bind;
 }
 
+/**
+ * Check if private broadcast can be done to clearnet peers and if yes via which proxy.
+ * If private broadcast connections should not be opened to `net`, then this will return an empty optional.
+ */
+static std::optional<Proxy> ProxyForClearnetPrivateBroadcast(Network clearnet)
+{
+    if (clearnet != NET_IPV4 && clearnet != NET_IPV6) {
+        return std::nullopt;
+    }
+    Proxy tor_proxy;
+    const std::string onion_arg{gArgs.GetArg("-onion", "")};
+    if (onion_arg != "" && onion_arg == "0" && GetProxy(NET_ONION, tor_proxy)) {
+        // Use the Tor proxy (if provided) for private broadcast connections to clearnet peers.
+        // Note: we check whether -onion is explicitly used because GetProxy(NET_ONION) may
```

Comment:
Updated.

Reply:
Hmm, maybe we can keep track of whether we have have successfully managed to connect to at least one `.onion` address via the `-proxy`, then we can assume that is a Tor proxy? How does that sound?

Reply:
> > Is there no better way of checking whether `-proxy` is an actual Tor proxy or just a regular SOCKS5 proxy?

> maybe we can keep track of whether we have have successfully managed to connect to at least one .onion address

Done in `e85cc59bad...057c79365c`, see https://github.com/bitcoin/bitcoin/pull/29415#issuecomment-2125213533, also dropped the change in `tor.md` because that is not needed anymore.

Looks better now, thank you!